feature_engine/selection/probe_features_selector.py

```diff
@@ -0,0 +1,163 @@
+from typing import List, Union, Dict
+
+import numpy as np
+import pandas as pd
+
+from sklearn.model_selection import cross_validate
+
+from feature_engine.dataframe_checks import check_X_y
+from feature_engine.selection.base_selector import BaseSelector, get_feature_importances
+
+
+from feature_engine._variable_handling.variable_type_selection import (
+    _find_all_variables,
+    _find_categorical_and_numerical_variables,
+)
+
+from feature_engine.selection._docstring import (
+    _cv_docstring,
+    _features_to_drop_docstring,
+    _fit_docstring,
+    _get_support_docstring,
+    _initial_model_performance_docstring,
+    _scoring_docstring,
+    _threshold_docstring,
+    _transform_docstring,
+    _variables_attribute_docstring,
+    _variables_numerical_docstring,
+)
+
+class ProbeFeaturesSelection(BaseSelector):
+    """
+
+
+    """
+    def __init__(
+        self,
+        estimator,
+        scoring: str = "roc_auc",
+        n_iter: int = 10,
+        seed: int = 0,
+        # TODO: Do we need confirm_variable given that this selector will not be used in a pipeline?
+        # TODO: Do we need the parameter because it is a param of BaseSelector?
+        confirm_variables: bool = False,
+    ):
+
+        super().__init__(confirm_variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.n_iter = n_iter
+        self.seed = seed
+
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Create three random feature. Find initial model performance.
+        Sort features by importance.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+        """
+        # check input dataframe
+        X, y = check_X_y(X, y)
+
+        # find numerical and categorical variables
+        self.variables = _find_categorical_and_numerical_variables(X, None)
```

Comment:
This transformer takes an sklearn estimator, so it can only work with numerical variables